### DISCUSSION TITLE
Issue with applying western diet media in species level model

### DISCUSSION THREAD

**Discussion by Mushahid2521** [micom-dev/micom#191] [General]
Hi there, 
I hope you are doing great. I am using Micom to design a cooperative bacterial consortia.

I have carveme to generate single species metabolic models and then combined them with micom to create the reference database. 
Then I was able to simulate the knockout analysis in a minimum media. However, when I tried to apply the western diet media on it I am getting the error in the bottom: 

I have used the following code snippet to achieve it: 
```
from micom import load_pickle
from micom.qiime_formats import load_qiime_medium

medium = load_qiime_medium("/content/western_diet_gut.qza")
medium.index = medium.reaction

com = load_pickle("/content/ncbi_species_models/sample_1.pickle")
com.medium = medium.flux
ko_own_db = com.knockout_taxa(fraction=0.8, method="change")
```

```
[/usr/local/lib/python3.10/dist-packages/micom/solution.py](https://localhost:8080/#) in crossover(community, sol, fluxes)
    262             com.constraints["objective_" + sp].ub = None
    263     if s is None:
--> 264         raise OptimizationError(
    265             "crossover could not converge (status = %s)." % community.solver.status
    266         )

OptimizationError: crossover could not converge (status = infeasible).
```

It would be much appreciated if you please help me on this. 
Thank you

**Comment by cdiener** [comment]
Hi the MICOM media from the media repository are usually specific for a given model database and metabolite ID system. For carveme you would need to use the version that has "carveme" in the name. Depending on your organisms this might not give you growth though and would need to be completed. You could have a look at the recipes at https://github.com/micom-dev/media for a description how this is done.
